### PR TITLE
[test-only] fix tests after auto-accepting enabling

### PR DESCRIPTION
**Pull request by ScharfViktor**
fix tests from https://github.com/owncloud/ocis/issues/7504

### CODE REVIEW

File: tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature [hunk 1]
```diff
@@ -423,7 +423,12 @@ Feature: sharing
     Then the OCS status code should be "<ocs_status_code>"
     And the HTTP status code should be "200"
     And the content of file "/Shares/lorem.txt" for user "Brian" should be "some content"
-    And the content of file "/Shares/lorem.txt" for user "Carol" should be "some content"
+    When the administrator adds user "Carol" to group "grp1" using the provisioning API
+    And user "Carol" should not see the following elements
+      | /Shares/lorem.txt |
+    And the sharing API should report to user "Carol" that these shares are in the pending state
+      | path       |
+      | /lorem.txt |
```
Comment (ScharfViktor) [inline tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:431]: interesting case, I think behavior is correct

File: tests/acceptance/features/coreApiShareOperationsToShares1/gettingShares.feature [hunk 2]
```diff
@@ -80,7 +80,7 @@ Feature: sharing
       | 2               | 200             |
 
   @smokeTest
-  Scenario Outline: resource cannot be reshared to resource owner
+  Scenario Outline: resource can be reshared to resource owner
```
Comment (saw-jan) [inline tests/acceptance/features/coreApiShareOperationsToShares1/gettingShares.feature:83]: so the behavior changed?

Comment (ScharfViktor) [reply tests/acceptance/features/coreApiShareOperationsToShares1/gettingShares.feature:83]: test was wrong:
```
  Scenario Outline: resource can be reshared to resource owner
    Given using OCS API version "<ocs_api_version>"
    And group "grp1" has been created
    And user "Carol" has been created with default attributes and without skeleton files
    And user "Carol" has been added to group "grp1"
    And user "Carol" has created folder "/shared"
    And user "Carol" has uploaded file with content "some data" to "/shared/shared_file.txt"
    And user "Carol" has shared folder "/shared" with user "Brian"
    And user "Brian" has accepted share "/shared" offered by user "Carol"
    And user "Brian" has shared folder "/Shares/shared" with group "grp1"
    # no need to accept this share as it is Carol's file
    When user "Carol" gets all the shares shared with her using the sharing API
    Then the OCS status code should be "<ocs_status_code>"
    And the HTTP status code should be "200"
    And the last share id should not be included in the response
    Examples:
      | ocs_api_version | ocs_status_code |
      | 1               | 100             |
      | 2               | 200             |
```
test passed but Carol had `/Shares/shared` in the pending state

how we can reshare to owner: 
1. `user1` shares to `user2`. `user2` reshares to `user1` - 400 `error creating share:error: bad request: jsoncs3: owner/creator and grantee are the same`
2. `user1` shares to `user2`. `user2` reshares to `group1`(`user1` is member) - worked and works now. without accepting-> `user1` has penndig share,  with accepting-> `user1` has accepted share,

File: tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature [hunk 3]
```diff
@@ -243,9 +243,9 @@ Feature: sharing
     And user "Brian" has been added to group "grp1"
     And user "Brian" has been added to group "grp2"
     And user "Alice" has created folder "/PARENT"
-    When user "Alice" shares folder "/PARENT" with group "grp1" using the sharing API 
+    When user "Alice" shares folder "/PARENT" with group "grp1" using the sharing API
     Then user "Brian" should see the following elements
-      | /Shares/PARENT |
+      | /Shares/PARENT/ |
```
[the remark this answers is not in the document]
Comment (ScharfViktor) [reply tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:248]: `/` is needed. without it doesn't work. see also line 232 220 217

[the remark this answers is not in the document]
Comment (saw-jan) [reply tests/acceptance/features/coreApiShareManagementBasicToShares/createShareToSharesFolder.feature:248]: hmm, then we must have a fix for that in the test code. But fine here

**Review by saw-jan — APPROVED** [review]
:+1:

**Pull request merged**